components/nimbus/src/enrollment.rs

```diff
@@ -185,6 +189,10 @@ impl ExperimentEnrollment {
                         self.disqualify_from_enrolled(DisqualifiedReason::Error);
                     out_enrollment_events.push(updated_enrollment.get_change_event());
                     updated_enrollment
+                } else if matches!(reason, EnrolledReason::OptIn) {
+                    // we check if we opted-in an experiment, if so
+                    // we don't need to update our enrollment
+                    self.clone()
```

Comment:
Hmmm I'm still slightly unsure about this, because if we go through the `evaluate_enrollment` codepath, shouldn't we still end up on the `self.clone()` in line 213?  Which means that the enrollment status and reason would have been preserved.

I still need to get familiarized a bit more with the code too, so I could be missing something